ixmp/core.py

```diff
@@ -48,23 +48,44 @@ def start_jvm(jvmargs=None):
 
 
 class Platform(object):
-    """ The class 'Platform' is the central access point to
-    the ix modeling platform (ixmp).
+    """Database-backed instance of the ixmp.
+
+    Each Platform connects three components:
+
+    1. A **database** for storing model inputs and outputs. This may either be
+       a local file (``dbtype='HSQLDB'``) or a database server accessed via a
+       network connection. In the latter case, connection information is read
+       from a `properties file`.
+    2. A Java Virtual Machine (**JVM**) to run core ixmp logic.
+    3. One or more GAMS **model(s)**.
```

Comment:
`Scenario.solve()` ends with these lines:
https://github.com/iiasa/ixmp/blob/92d3f4c4130ae1f54d558a36930054ebaacfaee8/ixmp/core.py#L1199-L1203

I can see how it would be pretty easily possible to swap these lines out for something else. But is it currently possible to solve a non-GAMS model? How?

I'll edit the text to say "models, implemented in GAMS or other languages/frameworks."

Reply:
One would have to implement other low-level interfaces to export data, execute and read the solution. The `solve()` function could be extended to start with an `if model_file.endswith('.gms')`, and then have call other low-level interfaces in the `elif` part.